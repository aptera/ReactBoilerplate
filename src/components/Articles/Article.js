import React from 'react';
import './styles/article.scss';

const Article = (props) => {
    const { article, actions } = props;
    const author = actions.lookupAuthor(article.authorId);
    return (
        <div className="article">
            <div className="h2">{article.title}</div>
            <div className="date">{article.date}</div>
            <div>
                <a className="website" href={author.website}>
                    {author.firstName} {author.lastName} GitHub Account
                </a>
            </div>
            <div className="body">{article.body}</div>
        </div>
    );
};

export default Article;