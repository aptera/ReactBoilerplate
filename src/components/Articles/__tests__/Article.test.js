import React from 'react';
import Article from '../Article';

import renderer from 'react-test-renderer';

describe('The Article', () => {
    const props = {
        article: {
            title: 'The Littlest Worm',
            data: '1/1/2011',
            body: 'The littlest worm you ever saw got stuck inside my soda straw.'
        },
        actions: {
            lookupAuthor: () => { 
                return {
                    website: 'http://www.author.com',
                    firstName: 'David',
                    lastName: 'Federspiel'
                };
            }
        }
    };

    it('renders correctly', () => {
        const tree = renderer.create(
            <Article
                {...props}
            />
        ).toJSON();
        expect(tree).toMatchSnapshot();
    });
});