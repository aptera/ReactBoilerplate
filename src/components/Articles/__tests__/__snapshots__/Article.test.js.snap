// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The Article renders correctly 1`] = `
<div
  className="article"
>
  <div
    className="h2"
  >
    The Littlest Worm
  </div>
  <div
    className="date"
  />
  <div>
    <a
      className="website"
      href="http://www.author.com"
    >
      David
       
      Federspiel
       GitHub Account
    </a>
  </div>
  <div
    className="body"
  >
    The littlest worm you ever saw got stuck inside my soda straw.
  </div>
</div>
`;
